### PR TITLE
Smol update

### PR DESCRIPTION
**Pull request by 0xdeafbeef**
It works much more better than version with smol 0.1. 
Need to rewrok part with blocking mode, but wraping session in unblock works as a charm.

### CODE REVIEW

File: src/session.rs [hunk 1]
```diff
@@ -51,7 +51,7 @@ impl Session {
         START.call_once(|| {
             std::thread::spawn(|| smol::run(futures::future::pending::<()>()));
         });
-        session.set_blocking(false);
+        session.set_blocking(true);
```
Comment (spebern) [inline src/session.rs]: This makes the whole session synchronous so it basically takes away the point of this library. Am I missing something?

**Review by spebern — COMMENTED** [review]
An update of smol requires more parts to be changed which I think should be done in a single commit. Otherwise in the case of stream this is good to go :).

**Comment by spebern** [comment]
Sadly the upgrade to smol 0.3 is not that trivial. Smol now uses the `read_with` and `write_with` methods. In cases where `AsyncWrite` and `AsyncRead` need to be implemented using them is trival. In other cases it is not so obvious if we need to wait for write or read events. For this `ssh::Session` provides the method [block_direction](https://docs.rs/ssh2/0.8.2/ssh2/struct.Session.html#method.block_directions).

So in order to listen for the right event the session needs to be accessible for other structs like `Agent`.

File: src/session.rs (hunk 1 above)
Comment (0xdeafbeef) [reply src/session.rs]: It was working, but not asynchronously:)
Fixed https://github.com/spebern/async-ssh2/issues/14, but awfully.

I've implemented better solution in https://github.com/0xdeafbeef/async-ssh2/blob/master/src/session.rs

**Pull request closed without merging**